android/app/src/main/java/com/microsoft/codepush/common/core/CodePushBaseCore.java

```diff
@@ -0,0 +1,1043 @@
+            }
+            case INSTALLING_UPDATE: {
+                AppCenterLog.info(LOG_TAG, "Installing update.");
+                break;
+            }
+            case UP_TO_DATE: {
+                AppCenterLog.info(LOG_TAG, "App is up to date.");
+                break;
+            }
+            case UPDATE_IGNORED: {
+                AppCenterLog.info(LOG_TAG, "User cancelled the update.");
+                break;
+            }
+            case UPDATE_INSTALLED: {
+                if (mState.mCurrentInstallModeInProgress == ON_NEXT_RESTART) {
+                    AppCenterLog.info(LOG_TAG, "Update is installed and will be run on the next app restart.");
+                } else if (mState.mCurrentInstallModeInProgress == ON_NEXT_RESUME) {
+                    AppCenterLog.info(LOG_TAG, "Update is installed and will be run after the app has been in the background for at least " + mState.mMinimumBackgroundDuration + " seconds.");
+                } else {
+                    AppCenterLog.info(LOG_TAG, "Update is installed and will be run when the app next resumes.");
+                }
+                break;
+            }
+            case UNKNOWN_ERROR: {
+                AppCenterLog.info(LOG_TAG, "An unknown error occurred.");
+                break;
+            }
+        }
+    }
+
+    /**
+     * Notifies listeners about changed update download progress.
+     *
+     * @param receivedBytes received amount of bytes.
+     * @param totalBytes    total amount of bytes.
+     */
+    protected void notifyAboutDownloadProgressChange(long receivedBytes, long totalBytes) {
+        for (CodePushDownloadProgressListener downloadProgressListener : mListeners.mDownloadProgressListeners) {
+            downloadProgressListener.downloadProgressChanged(receivedBytes, totalBytes);
+        }
+    }
+
+    /**
+     * Notifies listeners about binary version mismatch between local and remote packages.
+     *
+     * @param update remote package.
+     */
+    @SuppressWarnings("WeakerAccess")
+    protected void notifyAboutBinaryVersionMismatchChange(CodePushRemotePackage update) {
+        for (CodePushBinaryVersionMismatchListener listener : mListeners.mBinaryVersionMismatchListeners) {
+            listener.binaryVersionMismatchChanged(update);
+        }
+    }
+
+    /**
+     * Initializes update after app restart.
+     *
+     * @throws CodePushGetPackageException    if error occurred during the getting current package.
+     * @throws CodePushPlatformUtilsException if error occurred during usage of {@link CodePushPlatformUtils}.
+     * @throws CodePushRollbackException      if error occurred during rolling back of package.
+     */
+    @SuppressWarnings("WeakerAccess")
+    protected void initializeUpdateAfterRestart() throws CodePushGetPackageException, CodePushRollbackException, CodePushPlatformUtilsException {
+
+        /* Reset the state which indicates that
+         * the app was just freshly updated. */
```

Comment:
Is there a reason for the comment to be multiline?

Reply:
@annakocheshkova Nope